backend/trafficrules/src/main/java/com/coffeejug/trafficrules/service/UserService.java

```diff
@@ -0,0 +1,70 @@
+package com.coffeejug.trafficrules.service;
+
+import com.coffeejug.trafficrules.db.Progress;
+import com.coffeejug.trafficrules.db.User;
+import com.coffeejug.trafficrules.dto.UserDto;
+import com.coffeejug.trafficrules.projection.ProgressProjection;
+import com.coffeejug.trafficrules.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private ProgressService progressService;
+
+
+    public UserDto createNew() {
+
+        User user = new User();
+        user = userRepository.save(user);
+        UserDto userDto = new UserDto(user.getUuid().toString());
+        return userDto;
+    }
+
+    public List<ProgressProjection> getUserProgress(String uuid) {
+
+        if (goodUUID(uuid)) {

```

Comment:
If I take one symbol from 4th group (ab04a895-172b-491a-bc0-49838ca46c05) exeption isn't being thrown, because they check only count of groups:  if (components.length != 5)
            throw new IllegalArgumentException("Invalid UUID string: "+name);

Reply:
I wanna to check itf UUID is good before finding it in DB.

Reply:
It could be that this UUID is valid. Even if it's not, then the record won't be present in DB. So this logic is redundant and can be avoided to reduce code complexity

Reply:
But it makes extra DB call you tried to avoid in below conversation.

Reply:
I was trying to avoid not the extra DB call but redundant logic. Besides that, here it won't have any extra call, as if the UUID is valid we can't know if it's present in DB without calling the DB. So method `goodUUID` won't save us this call

Reply:
goodUUID saves us call to DB in case of wrong length UUID